src/biz/metrics.go

```diff
@@ -94,8 +94,9 @@ func (mc *MetricsFetcher) Process(containerID string) (Sample, error) {
 	if err != nil {
 		return metrics, err
 	}
+	// TODO: move logic to skip container without State to Docker specific code.
 	if json.State == nil {
-		return metrics, fmt.Errorf("invalid container %v JSON: missing State", json.ID)
+		log.Debug("invalid container %v JSON: missing State", json.ID)
```

Comment:
caller already knows about the container ID, so we could return a fixed error that's easier to handle by callers

Reply:
btw, why are you logging? shouldn't this prevent from fetching?

Reply:
It doesn't prevent fetching at all in Fargate. So far, 100% of the Fargate containers I've seen don't have a `State`, they only have a `Status`. 

I want to keep the container ID in the log message to allow it to make sense without requiring any context from previous log messages. The implementation of the caller might change and then this log message would be incomplete.